data/org.freedesktop.portal.Notification.xml

```diff
@@ -171,6 +171,21 @@
           * ``target`` (``v``)
 
             Target parameter to send along when activating the action.
+
+        * ``desktop-file-id`` (``s``)
```

Comment:
I really doubt this is a good idea. The portal should be able to determine the app id and fetch the desktop file based on that. What are you trying to accomplish here?

Reply:
This tries to solve the following:

- The current API can't really be used outside of flatpak, because there isn't a reliable why to get the app id. 
- A single flatpak may export multiple desktop files. There isn't any way how the portal can know which desktop-file-id should be used for D-Bus activation.

Reply:
A lot of things don't work (as well) when the app id can't be determined for non-flatpak apps. We should work towards making this detection mechanism more reliable.

> A single flatpak may export multiple desktop files and there isn't any way how the portal can know which desktop-file-id should be used for D-Bus activation, app name and icon.

That's a fair point. If we want to allow the app to choose which sub-desktop id (do we have a good name for this?) to use, this should probably be available to all portals, not specific to the notification portal.

Can we leave this out of the notification portal for now and open a new discussion for this issue?

Reply:
> Can we leave this out of the notification portal for now and open a new discussion for this issue?

I think it's fairly important, so apps can switch to this API. Especially if we go with https://gitlab.freedesktop.org/xdg/xdg-specs/-/issues/131

Reply:
btw the desktop-file-id isn't used for permission, it's meant for getting the right "app" that should be shown as the source for the notification.

Reply:
>  this should probably be available to all portals, not specific to the notification portal.

What other portals would need this?

Reply:
Right, I get the idea. For flatpak only desktop files starting with the app id would be allowed, with snap there isn't such a connection between the app id and desktop files.

There also isn't a check in place to make sure the backend only gets desktop file ids that are authentic.

Reply:
> There also isn't a check in place to make sure the backend only gets desktop file ids that are authentic.

What check would you like? If the desktop file exists?

Reply:
For flatpak the desktop files the app can export are all starting with the app id. For snap I have no idea. For host apps it would be nice to have the same logic as for flatpak but that might require apps to change their desktop file naming for host installs, so probably just accept anything.

Reply:
This is something you can delegate to XdpAppInfo. Give it a desktop file and it will return if it "belongs" to the app or not.

Reply:
We talked about it in the Matrix room and (I think?) we agreed on doing this delegate-to-XdpAppInfo change to a separate PR.